### PR TITLE
Update assistants-api-in-a-box

### PR DESCRIPTION
**Pull request by AndresPad**
Pull Request from FTA AI-in-a-Box Team: Neeraj Jhaveri, Alex Morales, Andres Padilla, Marco Cardoso

This pull request includes changes to the `assistant-bot-in-a-box` project. The changes mainly involve the addition of new files and configurations for the bot's deployment and operation. The most important changes include the addition of a `.gitignore` file, configuration files for the Visual Studio Code environment, a `README.md` file with comprehensive instructions and information about the bot, and several Azure Resource Manager (ARM) templates and parameters for deploying the bot to Azure.

Addition of new files and configurations:

* [`notebooks/assistant-bot-in-a-box/.gitignore`](diffhunk://#diff-031f6a700c0a72a7100c6b1c746a69a810e13a404f4de8fb83c7465fa8cd0d52R1-R7): Added a `.gitignore` file to exclude certain files and directories from the repository.

* `notebooks/assistant-bot-in-a-box/.vscode/launch.json` and `notebooks/assistant-bot-in-a-box/.vscode/tasks.json`: Added configuration files for the Visual Studio Code environment, including settings for launching and building the bot. [[1]](diffhunk://#diff-74cdda706e8711e2938e6fcd0768dea952cf18fec7d5cf1297c5356c05520928R1-R35) [[2]](diffhunk://#diff-7643c5890976a3da33c75be6dcb171a66ccef88641f7c5357d83756a12ea0c5aR1-R41)

* [`notebooks/assistant-bot-in-a-box/README.md`](diffhunk://#diff-af7d110dfefe1f56a61bd0765ade0410673698c2a9cdd491730cad0f1ae6a16cR1-R115): Added a `README.md` file with comprehensive instructions and information about the bot, including its architecture, prerequisites, deployment, and operation.

* [`notebooks/assistant-bot-in-a-box/azure.yaml`](diffhunk://#diff-d37e194fd17360bc50f5aa76f1e92663aec110f5dda91d53ecfce15c5d70ea0cR1-R10): Added a configuration file for deploying the bot to Azure.

* [`notebooks/assistant-bot-in-a-box/infra/abbreviations.json`](diffhunk://#diff-e8e4e67c660aac68aa3bfa8dc0a6925e422032f2184452dd21b7d7b6fe2386cdR1-R138): Added a JSON file with abbreviations for Azure resource types.

Addition of Azure Resource Manager (ARM) templates and parameters:

* [`notebooks/assistant-bot-in-a-box/infra/main.bicep`](diffhunk://#diff-fadb1d57e22c7e49cb07486122c42118837744c5e5f78fc4c5e01490c2cc29bfR1-R122): Added an ARM template for deploying the bot to Azure, including resources such as managed identity, OpenAI account, storage account, Cosmos DB account, App Service, and bot service.

* [`notebooks/assistant-bot-in-a-box/infra/main.parameters.json`](diffhunk://#diff-4cdb23bed09b30fded390e6bf417d40ebca2d5915d6918778cb77038f8ae2df4R1-R47): Added a JSON file with parameters for the ARM template.

* `notebooks/assistant-bot-in-a-box/infra/modules/appservice.bicep`, `notebooks/assistant-bot-in-a-box/infra/modules/bing.bicep`, and `notebooks/assistant-bot-in-a-box/infra/modules/botservice.bicep`: Added ARM templates for specific Azure resources, including App Service, Bing account, and bot service. [[1]](diffhunk://#diff-4d20bec4adae325a08198d7e63e64ea25c26dd0364cb32ab02bc77cad6e7a553R1-R144) [[2]](diffhunk://#diff-7c52cf7b5f5f525d975f11a4d43a96b94327ecb3a820b9f5dc59e1cf671b4de2R1-R18) [[3]](diffhunk://#diff-f558903dd67ad13374d9916f76749195cc6c33d35903e7c0c2458b3312dcfd59R1-R29)# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure-Samples/azureai-samples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: notebooks/assistants-api-in-a-box/notebooks/assistant-failed_banks.ipynb
Comment (kdestin) [inline notebooks/assistants-api-in-a-box/notebooks/assistant-failed_banks.ipynb:1]: Please format this notebook according to this template: https://github.com/Azure-Samples/azureai-samples/blob/main/notebooks/template.ipynb

File: notebooks/assistants-api-in-a-box/notebooks/assistant-math_tutor.ipynb
Comment (kdestin) [inline notebooks/assistants-api-in-a-box/notebooks/assistant-math_tutor.ipynb:1]: Please format this notebook according to this template: https://github.com/Azure-Samples/azureai-samples/blob/main/notebooks/template.ipynb

File: notebooks/assistants-api-in-a-box/notebooks/assistant-personal_finance.ipynb
Comment (kdestin) [inline notebooks/assistants-api-in-a-box/notebooks/assistant-personal_finance.ipynb:1]: Please format this notebook according to this template: https://github.com/Azure-Samples/azureai-samples/blob/main/notebooks/template.ipynb

File: notebooks/assistants-api-in-a-box/notebooks/assistant-wind_farm.ipynb
Comment (kdestin) [inline notebooks/assistants-api-in-a-box/notebooks/assistant-wind_farm.ipynb:1]: Please format this notebook according to this template: https://github.com/Azure-Samples/azureai-samples/blob/main/notebooks/template.ipynb

File: notebooks/assistant-bot-in-a-box/README.md
Comment (kdestin) [inline notebooks/assistant-bot-in-a-box/README.md:1]: Please start this README according to this template: https://github.com/Azure-Samples/azureai-samples/blob/main/notebooks/README-template.md

**Review by kdestin — CHANGES_REQUESTED** [review]
Hi, thank you for your contribution! Please take a look at our  [Contributing Guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md) , which walk through the expectations for PRs to this repository.

---


Install [dev requirements and enable `pre-commit`](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md#2-install-dev-dependencies). Then run `pre-commit run --all-files` from the root of the repository. 

This'll:

* Format the notebooks with black
* Remove unneeded metadata
* Surface some linting issues that need to be addressed.

---
 Please:
* Put your samples under a `notebooks/Assistants` directory. 
* Put each Jupyter notebook in it's own directory, alongside a README and the data files it uses

Your PR should roughly look like

```
notebooks/
└── Assistants
    ├── api-in-a-box
    │   ├── failed_banks
    │   │   ├── assistant-failed_banks.ipynb
    │   │   ├── data
    │   │   │   └── failed_banks.csv
    │   │   └── README.md
    │   ├── math_tutor
    │   │   ├── assistant-math_tutor.ipynb
    │   │   ├── data
    │   │   └── README.md
    │   ├── personal_finance
    │   │   ├── assistant-personal_finance.ipynb
    │   │   ├── data
    │   │   │   └── portfolio.csv
    │   │   └── README.md
    │   ├── README.md
    │   ├── requirements.txt
    │   └── wind_farm
    │       ├── assistant-wind_farm.ipynb
    │       ├── data
    │       │   └── turbines.csv
    │       └── README.md
    └── bot-in-a-box/
```

File: scenarios/Assistants/assistants-api-in-a-box/notebooks/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+__pycache__
+.env
+.python-version
+*_db
```
Comment (brandom-msft) [inline scenarios/Assistants/assistants-api-in-a-box/notebooks/.gitignore:4]: For the below suggestion, I've only removed the duplicates. From the comment in the parent .gitignore, seems .python-version is set per project - not sure what that means for it staying/going, here.
```suggestion
.python-version
*_db
```

File: notebooks/assistant-bot-in-a-box/readme_assets/Assistant API.textClipping
[the remark this answers is not in the document]
Comment (brandom-msft) [reply notebooks/assistant-bot-in-a-box/readme_assets/Assistant API.textClipping:1]: moved to:
scenarios/Assistants/assistant-bot-in-a-box/readme_assets/Assistant API.textClipping

File: scenarios/Assistants/assistant-bot-in-a-box/readme_assets/architecture.vsdx
Comment (brandom-msft) [inline scenarios/Assistants/assistant-bot-in-a-box/readme_assets/architecture.vsdx:1]: https://github.com/Azure-Samples/azureai-samples/pull/46#discussion_r1478939239

File: notebooks/assistant-bot-in-a-box/README.md
Comment (brandom-msft) [reply notebooks/assistant-bot-in-a-box/README.md:1]: File moved [here](https://github.com/Azure-Samples/azureai-samples/pull/46/files#diff-88f5faca95018595151a98e72e90442a69d4860919b5738137ec82bbb197bb3b)

File: notebooks/assistants-api-in-a-box/notebooks/assistant-wind_farm.ipynb
Comment (brandom-msft) [reply notebooks/assistants-api-in-a-box/notebooks/assistant-wind_farm.ipynb:1]: File moved [here](https://github.com/Azure-Samples/azureai-samples/pull/46/files#diff-4b2e3c1a327b5b24bb399cb1e0ba989d4e3b7a81f28260ddcb1314e4bbd023b7)

File: notebooks/assistants-api-in-a-box/notebooks/assistant-personal_finance.ipynb
Comment (brandom-msft) [reply notebooks/assistants-api-in-a-box/notebooks/assistant-personal_finance.ipynb:1]: File moved [here](https://github.com/Azure-Samples/azureai-samples/pull/46/files#diff-ebfbc7a150f113c82b04515302f043d1cd7fa2adb80f93478d879a479ef27e2f)

File: notebooks/assistants-api-in-a-box/notebooks/assistant-math_tutor.ipynb
Comment (brandom-msft) [reply notebooks/assistants-api-in-a-box/notebooks/assistant-math_tutor.ipynb:1]: File moved [here](https://github.com/Azure-Samples/azureai-samples/pull/46/files#diff-15eebad35d351a030884475a46a596ba8319e06d81d103b8352728505280dd1f)

File: notebooks/assistants-api-in-a-box/notebooks/assistant-failed_banks.ipynb
Comment (brandom-msft) [reply notebooks/assistants-api-in-a-box/notebooks/assistant-failed_banks.ipynb:1]: File moved [here](https://github.com/Azure-Samples/azureai-samples/pull/46/files#diff-a66fe31be1cf06c0db6f87872f7e1370bb00df6aa72a7b36219e129df5162813)

File: notebooks/assistants-api-in-a-box/notebooks/.gitignore
[the remark this answers is not in the document]
Comment (brandom-msft) [reply notebooks/assistants-api-in-a-box/notebooks/.gitignore:1]: Added [suggestion at new file location](https://github.com/Azure-Samples/azureai-samples/pull/46#discussion_r1479091289)

File: notebooks/assistant-bot-in-a-box/readme_assets/architecture.vsdx
[the remark this answers is not in the document]
Comment (kdestin) [reply notebooks/assistant-bot-in-a-box/readme_assets/architecture.vsdx:1]: Resolved by https://github.com/Azure-Samples/azureai-samples/pull/46/commits/461306ab8d07d3d31485dd87f2ebcfd61042ee37

File: notebooks/assistants-api-in-a-box/readme_assets/assistantsapi-flow-overview.png
[the remark this answers is not in the document]
Comment (kdestin) [reply notebooks/assistants-api-in-a-box/readme_assets/assistantsapi-flow-overview.png:1]: Resolved by https://github.com/Azure-Samples/azureai-samples/pull/46/commits/461306ab8d07d3d31485dd87f2ebcfd61042ee37

File: notebooks/assistant-bot-in-a-box/readme_assets/Assistant API.textClipping
[the remark this answers is not in the document]
Comment (kdestin) [reply notebooks/assistant-bot-in-a-box/readme_assets/Assistant API.textClipping:1]: https://github.com/Azure-Samples/azureai-samples/pull/46/commits/461306ab8d07d3d31485dd87f2ebcfd61042ee37

**Comment by kdestin** [comment]
Will be followed by a PR that addresses unaddressed comments in this 

cc: @AndresPad

**Pull request merged**